### ISSUE TITLE
In a user stack service, host ports are being reused when increasing in scale

### ISSUE THREAD

**Issue by n0rig** [rancher/rancher#7400]
**Rancher Versions:**
Server: v1.3.0
healthcheck: v0.2.0
ipsec: 
network-services: v0.8.1
scheduler: v0.5.0
kubernetes (if applicable):

**Docker Version:**
v1.12.5

**OS and where are the hosts located? (cloud, bare metal, etc):**
centos

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
1.) create a new user stack
2.) Add a new service to the stack with just one primary service container.
3.) Add port mappings...but only fill out the "Private Container Port" sections. Not the "Public Host Port" sections. This allows docker to do it's own host->container mapping.
4.) save the server.
5.) In the service page, bump the "scale" to 2 or higher.

**Results:** 
The rancher service manager stops the previously running container and restarts it and creates the necessary extra containers in to match the "scale" you want. BUT, the problem is, all of the NEW containers created get assigned the same PUBLIC HOST PORTS. Thus, the new containers clash with respect to port assignments and they cannot be started.

For example (after increasing the scale number):
service-container-1:   2342->80, 35242->22. 5345->23
service-container-2:   2342->80, 35242->22, 5345->23

**Expected:**
If I run the same containers via CLI, by using "-p 80 -p 22 -p 23" (etc), and not "-p 80:80 -p 22:22 -p 23:23", I can start multiple containers of the same application. The good thing about this is that I can reach the containers independently by their hostnames/ip addresses.

But, the problem is that rancher/cattle are FORCING the same Public Host Port mappings across all my service instances...and I do not want that...that's why I left them blank in the service configuration page. Each container needs to get a new host port mapping.

For example:
service-container-1:   2342->80, 35242->22. 5345->23
service-container-2:   5435->80, 5653->22, 1235->23
service-container-3:   5753->80, 7869->22, 9123->23

**Comment by vincent99** [comment]
https://github.com/rancher/rancher/issues/4178

TL;DR: the same port is intentionally used but switching to a new number when no more are available is missing.
